llmfoundry/data/finetuning/collator.py

```diff
@@ -398,18 +398,18 @@ def _process_and_batch_decoder_only(
             return_tensors='pt',
         )
 
-        batch['sequence_id'] = batch['attention_mask'] - 1
+        batch['sequence_id'] = batch['attention_mask'] - 1  # type: ignore
```

Comment:
nit: to get rid of these type ignores, you could do 
```
batch: Dict[str, torch.Tensor] = self.tokenizer.pad(
            processed_examples,
            padding='max_length',
            max_length=max_seq_len,
            return_tensors='pt',
        )
```

above

Reply:
thats technically not the type though...its a `BatchEncoding` which inherits `UserDict`, so behaves as a dict. im just gonna leave it. complying with the tokenizer output types is pretty messy.